lib/ood_core/batch_connect/template.rb

```diff
@@ -68,6 +70,10 @@ def to_s
         EOT
       end
 
+      def passwd_size
+        context.fetch(:passwd_size, DEFAULT_PASSWD_SIZE).to_i
```

Comment:
Typically it is best practice to do as you have done: replace magic literals with a constant that describes what it is. However, in the case of Ruby code where we have attr accessors or methods that return a custom value OR a default value, it is clear what the magic literal is and so I'm not sure that taking the extra step to add the does not add more clarity.

These are examples of code where replacing with a constant doesn't add anything IMHO:

```ruby
# 8 is obviously the default password size
context.fetch(:passwd_size, 8).to_i

# 2000 is obviously default_min_port
min_port    = context.fetch(:min_port, 2000).to_i

# "/etc/ood/config/apps/dashboard" is obviously the default config_root
def config_root
  Pathname.new(ENV["OOD_APP_CONFIG_ROOT"] || "/etc/ood/config/apps/dashboard")
end

# "development" is obviously the default rails_env
def rails_env
  ENV['RAILS_ENV'] || ENV['RACK_ENV'] || "development"
end
```

But obviously in cases where it isn't clear, like when the magic literal is an operand in an expression, it definitely calls for it. Also, perhaps when the same literal is used in multiple places.